File: app/src/main/java/com/paradoxo/amadeus/dao/AcaoDAO.java
package com.paradoxo.amadeus.dao;

import android.content.Context;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.paradoxo.amadeus.modelo.Acao;

import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.Charset;
import java.util.Arrays;
import java.util.List;
import java.util.Objects;

public class AcaoDAO {
    public static final String ARQ_LISTA_ACOES_JSON = "ListaAcoes.json";
    Context context;

    public AcaoDAO(Context context) {
        this.context = context;
    }

    public List<Acao> getAcoes() {
        String json;
        InputStream inputStream;
        try {
            inputStream = Objects.requireNonNull(context).getAssets().open(ARQ_LISTA_ACOES_JSON);
            int tamanho = inputStream.available();
            byte[] buffer = new byte[tamanho];
            inputStream.read(buffer);
            inputStream.close();
            json = new String(buffer, Charset.forName("UTF-8"));
        } catch (IOException e) {
            e.printStackTrace();
            return null;
        }

        Gson gson = new GsonBuilder().create();

        return Arrays.asList(gson.fromJson(json, Acao[].class));
    }

    public Long getQuantidadeTotal() {
        String json;
        InputStream inputStream;
        try {
            inputStream = Objects.requireNonNull(context).getAssets().open(ARQ_LISTA_ACOES_JSON);
            int tamanho = inputStream.available();
            byte[] buffer = new byte[tamanho];
            inputStream.read(buffer);
            inputStream.close();
            json = new String(buffer, Charset.forName("UTF-8"));
        } catch (IOException e) {
            e.printStackTrace();
            return null;
        }

        Gson gson = new GsonBuilder().create();

        return (long) Arrays.asList(gson.fromJson(json, Acao[].class)).size();
    }
}

File: app/src/main/java/com/paradoxo/amadeus/dao/AcaoDAO.kt
package com.paradoxo.amadeus.dao

import android.content.Context
import com.google.gson.GsonBuilder
import com.paradoxo.amadeus.modelo.Acao

class AcaoDAO(private val context: Context) {
    companion object {
        const val ARQ_LISTA_ACOES_JSON = "ListaAcoes.json"
    }

    private fun lerJson(): String? {
        return try {
            val inputStream = context.assets.open(ARQ_LISTA_ACOES_JSON)
            val buffer = ByteArray(inputStream.available())
            inputStream.read(buffer)
            inputStream.close()
            String(buffer, Charsets.UTF_8)
        } catch (e: Exception) {
            e.printStackTrace()
            null
        }
    }

    fun getAcoes(): List<Acao>? {
        val json = lerJson() ?: return null
        return GsonBuilder().create().fromJson(json, Array<Acao>::class.java).toList()
    }

    val quantidadeTotal: Long
        get() = getAcoes()?.size?.toLong() ?: 0L
}
